cfme/services/myservice.py

```diff
@@ -89,11 +161,16 @@ def wait_for_vm_retire(self):
         Returns: A string representing the VM state
         """
         def get_vm_details():
-            power_state = InfoBlock.text('Power Management', 'Power State')
+            power_details = SummaryTable(title='Power Management')
```

Comment:
Nope, this should not work as definitions like this work only when defined on widget class. You should either define it somewhere in this module or if it is from some other module (VM view), then that should be used and instantiated. If none such exists, you should create some provisional view with these widgets which should have TODO to replace them with correct references.